### PR TITLE
backport arm64 build and rocksdb build support in makefile

### PR DESCRIPTION
**Pull request by allthatjazzleo**
1. go 1.17 in goreleaser ci
2. arm64 support release
3. rocksdb support in the makefile
4. update go.mod to support arm64 and rocksdb
5. fix go module v1 issue

# PR Checklist:

- [ ] Have you read the [CONTRIBUTING.md](https://github.com/crypto-org-chain/chain-main/blob/master/CONTRIBUTING.md)?
- [ ] Does your PR follow the [C4 patch requirements](https://rfc.zeromq.org/spec:42/C4/#23-patch-requirements)?
- [ ] Have you rebased your work on top of the latest master? 
- [ ] Have you checked your code compiles? (`make`)
- [ ] Have you included tests for any non-trivial functionality?
- [ ] Have you checked your code passes the unit tests? (`make test`)
- [ ] Have you checked your code formatting is correct? (`go fmt`)
- [ ] Have you checked your basic code style is fine? (`golangci-lint run`)
- [ ] If you added any dependencies, have you checked they do not contain any known vulnerabilities? (`go list -json -m all | nancy sleuth`)
- [ ] If your changes affect the client infrastructure, have you run the integration test?
- [ ] If your changes affect public APIs, does your PR follow the [C4 evolution of public contracts](https://rfc.zeromq.org/spec:42/C4/#26-evolution-of-public-contracts)?
- [ ] If your code changes public APIs, have you incremented the crate version numbers and documented your changes in the [CHANGELOG.md](https://github.com/crypto-org-chain/chain-main/blob/master/CHANGELOG.md)?
- [ ] If you are contributing for the first time, please read the agreement in [CONTRIBUTING.md](https://github.com/crypto-org-chain/chain-main/blob/master/CONTRIBUTING.md) now and add a comment to this pull request stating that your PR is in accordance with the [Developer's Certificate of Origin](https://github.com/crypto-org-chain/chain-main/blob/master/CONTRIBUTING.md#developer-certificate-of-originn).

Thank you for your code, it's appreciated! :)

### CODE REVIEW

**Review by tomtau — CHANGES_REQUESTED** [review]
the changelog could be updated + this may need backporting some changes to nix integration tests etc.?

**Comment by allthatjazzleo** [comment]
@yihuang 
Could you help on nix integration tests

**Comment by yihuang** [comment]
The nix Ci error message:
```
error: experimental Nix feature 'nix-command' is disabled; use '--extra-experimental-features nix-command' to override
```

It looks like you need to upgrade these actions:
```
      - uses: cachix/install-nix-action@v15
      - uses: cachix/cachix-action@v10
```

**Comment by tomtau** [comment]
```
nix run -f ./. run-integration-tests -c run-integration-tests
error: unrecognised flag '-c'
```

I guess the nix scripts may need further fixes?

**Comment by yihuang** [comment]
> ```
> nix run -f ./. run-integration-tests -c run-integration-tests
> error: unrecognised flag '-c'
> ```
> 
> I guess the nix scripts may need further fixes?

I just tried locally, it seems we can simplely remove the `-c run-integration-tests` part:
```
nix run -f ./. run-integration-tests
```

**Comment by yihuang** [comment]
I'm trying to fix it in another PR: https://github.com/crypto-org-chain/chain-main/pull/695 , not succeed yet.

**Comment by allthatjazzleo** [comment]
btw, I revert the backport commit due to apphash mismatch during syncing
```
INF ABCI Replay Blocks appHeight=303290 module=consensus stateHeight=303290 storeHeight=303291
INF Replay last block using real app module=consensus
Error: error during handshake: error on replay: wrong Block.Header.AppHash.  Expected C98B1B8154A3C70D1894F54CBC17DAA55DD93806D1BE2B6CF585705A1F490D0D, got 8B9073C005A98CB77A94CAB0744C0BA8A35ED9A633BB7BBF07CD5BA2AF4E91A4
```
as the original cosmos-sdk in 1.2.1 can sync through this block

**Comment by tomtau** [comment]
@allthatjazzleo can you try to rebase on top of the latest one?

File: CHANGELOG.md [hunk 1]
```diff
@@ -2,6 +2,17 @@
 
 *unreleased*
 
+## v1.2.2
+A version based on the upstream release of Cosmos SDK 0.42.11.
+
```
Comment (tomtau) [inline CHANGELOG.md:7]:
```suggestion
*January 27, 2022*

## v1.2.2
  This release is identical to v1.1.1, but is built with Go 1.17 and supports Arm64 binaries as well as a RocksDB database backend.
```

**Pull request merged**